openscm/scmdataframe/base.py

```diff
@@ -0,0 +1,810 @@
+        kwargs:
+            Other arguments to pass through to `interpolate`
+
+        Examples
+        --------
+
+            # resample a dataframe to annual values
+        >>> scm_df = ScmDataFrame(pd.Series([1, 2, 10], index=(2000, 2001, 2009)), columns={
+            "model": ["a_iam"],
+            "scenario": ["a_scenario"],
+            "region": ["World"],
+            "variable": ["Primary Energy"],
+            "unit": ["EJ/y"],}
+        )
+        >>> scm_df.timeseries().T
+        model             a_iam
+        scenario     a_scenario
+        region            World
+        variable Primary Energy
+        unit               EJ/y
+        year
+        2000                  1
+        2010                 10
+
+        An annual timeseries can be the created by interpolating between to the start of years using the rule 'AS'.
+
+        >>> res = scm_df.resample('AS')
+        >>> res.timeseries().T
+        model                        a_iam
+        scenario                a_scenario
+        region                       World
+        variable            Primary Energy
+        unit                          EJ/y
+        time
+        2000-01-01 00:00:00       1.000000
+        2001-01-01 00:00:00       2.001825
+        2002-01-01 00:00:00       3.000912
+        2003-01-01 00:00:00       4.000000
+        2004-01-01 00:00:00       4.999088
+        2005-01-01 00:00:00       6.000912
+        2006-01-01 00:00:00       7.000000
+        2007-01-01 00:00:00       7.999088
+        2008-01-01 00:00:00       8.998175
+        2009-01-01 00:00:00      10.00000
+
+        >>> m_df = scm_df.resample('MS')
+        >>> m_df.timeseries().T
+        model                        a_iam
+        scenario                a_scenario
+        region                       World
+        variable            Primary Energy
+        unit                          EJ/y
+        time
+        2000-01-01 00:00:00       1.000000
+        2000-02-01 00:00:00       1.084854
+        2000-03-01 00:00:00       1.164234
+        2000-04-01 00:00:00       1.249088
+        2000-05-01 00:00:00       1.331204
+        2000-06-01 00:00:00       1.416058
+        2000-07-01 00:00:00       1.498175
+        2000-08-01 00:00:00       1.583029
+        2000-09-01 00:00:00       1.667883
+                                    ...
+        2008-05-01 00:00:00       9.329380
+        2008-06-01 00:00:00       9.414234
+        2008-07-01 00:00:00       9.496350
+        2008-08-01 00:00:00       9.581204
+        2008-09-01 00:00:00       9.666058
+        2008-10-01 00:00:00       9.748175
+        2008-11-01 00:00:00       9.833029
+        2008-12-01 00:00:00       9.915146
+        2009-01-01 00:00:00      10.000000
+        [109 rows x 1 columns]
+
+
+        Note that the values do not fall exactly on integer values due the period between years is not exactly the same
```

Comment:
```suggestion
        Note that the values do not fall exactly on integer values as the period between years is not exactly the same
```

Reply:
how hard would it be to offer a '365 day AS' for people who don't want this complication?